backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/BaseManagedBranchSnapshot.java

```diff
@@ -40,7 +41,7 @@
    * parent data somewhere outside of this class (e.g. in {@link GitMacheteRepositorySnapshot}).
    */
   protected void setParentForChildren() {
-    for (var child : children) {
+    for (val child : children) {
```

Comment:
Changing from `var` to `lombok.val` is 90% of this PR; the other changes are about some APIs added to Java after 8